### PR TITLE
Make possible to index and ignore preferences programatically

### PR DESCRIPTION
**Pull request by BrayanDSO**
Hi there! Here's the PR I said I would do.

This aims to solve two problems of mine:
- Index preferences programatically: as some are built programatically on AnkiDroid instead of on the XMLs
- Ignore preferences programatically: as some can't be used by everyone, e.g. newer API features and device specific settings (e.g. E-ink)

Please feel free to push back about the implementation.

I think I won't need any more changes to the library, so you can release a new version after this is merged. Pinging @ByteHamster just in case you haven't got the notification. Sorry to bother.

### CODE REVIEW

**Review by ByteHamster — COMMENTED** [review]
Thanks! I like the code structure. There are a few code style changes that I would ask you to do, though.

File: lib/src/main/java/com/bytehamster/lib/preferencesearch/PreferenceItem.java [hunk 1]
```diff
@@ -22,6 +25,25 @@ class PreferenceItem extends ListItem {
     private float lastScore = 0;
     private String lastKeyword = null;
 
+    PreferenceItem() { super(); }
```
[the remark this answers is not in the document]
Comment (BrayanDSO) [reply lib/src/main/java/com/bytehamster/lib/preferencesearch/PreferenceItem.java]: That's true. Thanks.

**Comment by BrayanDSO** [comment]
Thanks for taking the time to review this. I've done the changes. Happy to do any others if necessary.

**Comment by ByteHamster** [comment]
Thanks!

**Pull request merged**